transhumandesign/kag-base
# [added] Team-colored saddle to Bison

## Status

- **IN DEVELOPMENT**: this PR is still in development, but you would like your changes reviewed.

Will set to **READY** after addressing these:
- Maybe remove friend_property entry from AnimalConsts.as and AnimalBrain.as and change checks accordingly.
- Maybe make foods lying around forget DamageOwnerPlayer after a while

## Description

[added] Team-colored Saddle to Bison
[changed] Bison will remember the team of the player who tamed it instead of remembering only the taming player

This was the previous Bison logic:
* Remembers player who tamed it (tamer)
* If tamer hits Bison, Bison unfriends tamer without getting mad
* Will not hurt any players in tamer's team
* Forgets tamer when tamer blob disappears (going spectator, DC, possibly on death, too)
* Tamer can ride Bison as long as he wants

This PR changes the logic to this:
* Doesn't remember tamer specifically but remembers tamer's team - while this is true, carries a team-colored saddle
* If a player in tamer's team hits Bison, Bison unfriends the tamer's team without getting mad
* Will not hurt any players in tamer's team
* Doesn't forget tamer's team regardless who connects or disconnects
* Anyone in tamer's team can ride Bison as long as they want

https://i.imgur.com/TJIYV95.png
https://i.imgur.com/dhoteQY.png

I tried to make a good saddle sprite. Let me know how it can be improved.

## Steps to Test or Reproduce

Go to Sandbox.
Spawn a Bison.
Throw a food at it.
Notice it gains a team-colored Saddle.
Change team and do it again.
Notice the Saddle will change color.
Hurt the Bison.
Notice the saddle disappears.
